src/data/polynomial/degree/definitions.lean

```diff
@@ -222,6 +222,13 @@ begin
   { rw [← C_mul_X_pow_eq_monomial, nat_degree_C_mul_X_pow i r hr] }
 end
 
+lemma nat_degree_monomial_eq (i : ℕ) {r : R} (r0 : r ≠ 0) :
+  (monomial i r).nat_degree = i :=
+begin
+  classical,
+  exact eq.trans (nat_degree_monomial _ _) (if_neg r0),
```

Comment:
Minor golf: can you use a single `rw` block instead?

Reply:
I can, but the proof term is twice as long: I tend to prefer `eq.trans` when the rewrite applies to the whole lhs/rhs.